src/libraries/System.Runtime/tests/System/DoubleTests.GenericMath.cs

```diff
@@ -1387,6 +1458,77 @@ public static void TryWriteExponentLittleEndianTest()
             Assert.Equal(new byte[] { 0x00, 0x04 }, destination.ToArray());
         }
 
+        [Fact]
+        public static void TryWriteSignificandBigEndianTest()
+        {
+            Span<byte> destination = stackalloc byte[8];
+            int bytesWritten = 0;
+
+            Assert.True(FloatingPointHelper<double>.TryWriteSignificandBigEndian(double.NegativeInfinity, destination, out bytesWritten));
+            Assert.Equal(8, bytesWritten);
+            Assert.Equal(new byte[] { 0x00, 0x10, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 }, destination.ToArray());
```

Comment:
FWIW: We have an `AssertExtensions.SequenceEquals<T>(ROSpan<T>, ROSPan<T>)` for spans, if you don't like all the ToArray calls in your tests.

https://github.com/dotnet/runtime/blob/main/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs#L435-L476

Reply:
Good to know for future changes. I might do that as a later cleanup. As for now, the current logic works and allows easily copying and tweaking logic between the 15-20 types that need it.